skbeam/core/roi.py

```diff
@@ -430,18 +476,27 @@ def circular_average(image, calibrated_center, threshold=0, nx=100,
     ring_averages : array
         Radial average of the image. shape is (nx, ).
     """
-    radial_val = utils.radial_grid(calibrated_center, image.shape, pixel_size)
-
-    bin_edges, sums, counts = utils.bin_1D(np.ravel(radial_val),
-                                           np.ravel(image), nx,
+    radial_val = utils.radial_grid(calibrated_center, image.shape, pixel_size) 
+    if mask is not None:  
+        mask = np.array( mask, dtype = bool)
+        binr = radial_val[mask]
+        image_mask =     np.array( image )[mask]        
+    else:        
+        binr = np.ravel( radial_val ) 
+        image_mask = np.ravel(image) 
+    binr_ =   binr /(np.sqrt(pixel_size[1]*pixel_size[0] ))
```

Comment:
I agree with @CJ-Wright , I would also prefer to let the geometric mean be done by the user after the circular average and try to keep this data as raw as possible. (It's just a constant factor to be divided after the fact) 

So I also want to know the reasoning? It can't just be set to (1,1) to by pass since you need it to map out the radial_average code. As a user I vote for arithmetic :-P

thanks!